Automattic/wp-calypso
# Payment Methods: Add tax fields to PayPal

#### Changes proposed in this Pull Request

When assigning a payment method to a purchase, one of the available choices is (usually) PayPal. This creates a new PayPal billing agreement and assigns it to the purchase. However, there is no postal code or country code associated with the payment method and we need that for tax purposes.

This PR modifies the payment method to be able to collect tax information on its own.

Currently the form will still submit successfully with empty tax information. This is intentional; when we decide we are ready, the `/payment-methods/create-paypal-agreement` endpoint can be updated to return a validation error if the fields are not set, which this code will display to the user.

Before:

<img width="619" alt="Screen Shot 2022-02-23 at 8 07 43 PM" src="https://user-images.githubusercontent.com/2036909/155437856-d6fca932-f97f-4837-8ba0-bada01cc1135.png">

After, with postal code:

<img width="620" alt="Screen Shot 2022-02-23 at 8 13 35 PM" src="https://user-images.githubusercontent.com/2036909/155438212-e6a8341f-743e-4d44-bbfc-e9744fff8c56.png">

After, without postal code:

<img width="618" alt="Screen Shot 2022-02-23 at 8 09 57 PM" src="https://user-images.githubusercontent.com/2036909/155437905-290529ad-c981-430f-b398-413c510bd7fb.png">

Part of 549-gh-Automattic/payments-shilling

#### Testing instructions

First, verify that this does not change how PayPal is handled in checkout:

- Add a product to your cart and visit checkout.
- Select "PayPal" as the payment method.
- Verify that you do not see any tax information displayed on the payment method.
- Click the PayPal submit button.
- Verify that you are redirected to PayPal (no need to complete the purchase).

Then test the actual new behavior. You must use an account with at least one active purchase.

- Visit `/me/purchases` and select the purchase.
- Click "Add payment method" or "Change payment method".
- Select "PayPal" in the form that follows.
- Fill in tax information, but use invalid data. For example, a postal code of `ABCDEFG`.
- Click the PayPal submit button.
- Verify that you see a validation error.
- Correct the tax information and resubmit the form.
- Verify that you are redirected to PayPal (no need to complete the purchase).
